src/compiler/parser.ts

```diff
@@ -3015,7 +3015,11 @@ namespace ts {
                 let newPrecedence = getBinaryOperatorPrecedence();
 
                 // Check the precedence to see if we should "take" this operator
-                if (newPrecedence <= precedence) {
+                if (token === SyntaxKind.AsteriskAsteriskToken && newPrecedence < precedence) {
+                    // ** operator is right-assocative
+                    break;
+                }
+                else if (token !== SyntaxKind.AsteriskAsteriskToken && newPrecedence <= precedence) {

```

Comment:
I don't think the above check is the same as 
```typescript
if(newPrecenden <= precendence) {
    break;
}
```
you want to also break if the operator has lower precedence not just equal.
```typescript
a * b - c
```
will be parsed incorrectly as a * (b-c)